kornia/enhance/adjust.py

```diff
@@ -430,46 +462,35 @@ def equalize(input: torch.Tensor) -> torch.Tensor:
     """
     input = _to_bchw(input) * 255
 
-    # Code taken from: https://github.com/pytorch/vision/pull/796
-    def scale_channel(im, c):
-        """Scale the data in the channel to implement equalize."""
-        im = im[c, :, :]
-        # Compute the histogram of the image channel.
-        histo = torch.histc(im, bins=256, min=0, max=255)
-        # For the purposes of computing the step, filter out the nonzeros.
-        nonzero_histo = torch.reshape(histo[histo != 0], [-1])
-        step = (torch.sum(nonzero_histo) - nonzero_histo[-1]) // 255
-
-        def build_lut(histo, step):
-            # Compute the cumulative sum, shifting by step // 2
-            # and then normalization by step.
-            lut = (torch.cumsum(histo, 0) + (step // 2)) // step
-            # Shift lut, prepending with 0.
-            lut = torch.cat([torch.zeros(1), lut[:-1]])
-            # Clip the counts to be in range.  This is done
-            # in the C code for image.point.
-            return torch.clamp(lut, 0, 255)
-
-        # If step is zero, return the original image.  Otherwise, build
-        # lut from the full histogram and step and then index from it.
-        if step == 0:
-            result = im
-        else:
-            # can't index using 2d index. Have to flatten and then reshape
-            result = torch.gather(build_lut(histo, step), 0, im.flatten().long())
-            result = result.reshape_as(im)
-
-        return result / 255.
-
     res = []
     for image in input:
         # Assumes RGB for now.  Scales each channel independently
         # and then stacks the result.
-        scaled_image = torch.stack([scale_channel(image, i) for i in range(len(image))])
+        scaled_image = torch.stack([_scale_channel(image[i, :, :]) for i in range(len(image))])
         res.append(scaled_image)
     return torch.stack(res)
 
 
+def equalize3d(input: torch.Tensor):
```

Comment:
missing return type in definition `def equalize3d(input: torch.Tensor) -> torch.Tensor:`